challenge.rb

```diff
@@ -1,26 +1,30 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map{ |word| word.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  #normally this would be a select, but this is working around the "only once" constraint
+  input.each_with_object([]){ |word, ary| ary << word if word.downcase == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject{ |word| word.downcase == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.slice(0..1)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.select do |word|
+    initial_letter = word[0].downcase
+    initial_letter == "c" || initial_letter == "d"
```

Comment:
This would be more flexible if you extracted the comparison letters "C" and "D" out of the select. As it stands, this can only be used for magic strings "C" and "D". A simple solution would be to assign these to local variables, just so they can easily be changed.